src/components/button/button.tsx

```diff
@@ -1,12 +1,33 @@
-import React, { InputHTMLAttributes } from 'react';
-import { StitcheButton } from './styles';
+import React from 'react';
+import { ButtonStyles } from './styles';
 
-export interface ButtonProps extends InputHTMLAttributes<HTMLButtonElement> {
+type StitchesButtonProps = React.ComponentProps<typeof ButtonStyles>;
+
+export type ButtonProps = {
   label: string;
-}
+  type?: string;
+  disabled?: boolean;
+  onClick?: () => void;
+  isDanger?: boolean;
+} & StitchesButtonProps;
 
-const Button = ({ label }: ButtonProps) => {
-  return <StitcheButton>{label}</StitcheButton>;
+const IonButton = ({
+  type = 'primary',
+  disabled = false,
+  isDanger = false,
+  onClick,
```

Comment:
I believe that we can change to handleClick to not cause confusing with the prop name.

```suggestion
  handleClick,
```